app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -300,4 +308,22 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
                 serviceLocator.screenController.dispatchKeyEvent(event, fragmentManager) ||
                 super.dispatchKeyEvent(event)
     }
+
+    private fun observeReceivedTabs(): Disposable {
+        return serviceLocator.fxaRepo.receivedTabs
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { consumableTab ->
+            consumableTab.consume { tab ->
+                TelemetryIntegration.INSTANCE.receivedTabEvent(tab.metadata)
+                openReceivedFxaTab(tab)
+                true // Consume value
+            }
+        }
+    }
+
+    private fun openReceivedFxaTab(receivedTab: FxaReceivedTab) {
```

Comment:
nit: probably not worth the time to change but this might be a better inner function